src/Compilers/CSharp/Test/Symbol/Symbols/Source/LocalTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
+
+public class LocalTests : SemanticModelTestBase
+{
+    [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+    [Fact]
+    public void ForEach()
+    {
+        var sourceCode = @"
+class C
+{
+    void M()
+    {
+        var a = new int[] { 1, 2, 3 };
+        foreach (var x in a)
+        {
+            var y = /*<bind>*/x/*</bind>*/;
+        }
+    }
+}
+";
+        var compilation = CreateCompilation(sourceCode);
+        var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(compilation);
+
+        var symbol = (ILocalSymbol)semanticInfo.Symbol;
+        Assert.False(symbol.IsUsing);
+        Assert.True(symbol.IsForEach);
+        Assert.False(symbol.IsForEach()); // VB extension method works only for VB locals
+    }
+
+    [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+    [Fact]
+    public void ForEachVisualBasic()
+    {
+        var sourceCode = @"
+Module M1
+    Class C
+        Sub S() As Object
+            Dim a() As Integer = {1, 2, 3}
+            For Each x As Integer In a
+                Dim y = x   'BIND:""x""
+            Next
+        End Sub
+    End Class
+End Module
+";
+        var compilation = CreateVisualBasicCompilation(sourceCode);
+        var model = compilation.GetSemanticModel(compilation.SyntaxTrees[0]);
+        var semanticInfo = VisualBasic.UnitTests.CompilationUtils.FindBindingText<VisualBasic.Syntax.IdentifierNameSyntax>(compilation);
```

Comment:
>semanticInfo [](http://example.com/codeflow?start=12&length=12)

The name is somewhat misleading. I assume this is an identifier syntax.

Reply:
Good catch. Changed to be consistent with the VB tests: 2066c049609c371ad990e8d5c4e8930f5f323c80